src/profile-logic/processed-profile-versioning.js

```diff
@@ -595,5 +595,35 @@ const _upgraders = {
       }
     }
   },
+  [13]: profile => {
+    // The stackTable has a new column called "category", which is computed
+    // from the stack's frame's category, or if that is null, from the stack's
+    // prefix's category. For root stacks whose frame doesn't have a category,
+    // the category is set to the grey category (usually something like "Other").
+    // The same algorithm is used in profile processing, when the processed
+    // profile's category column is derived from the gecko profile (which does
+    // not have a category column in its stack table).
+    const { meta, threads } = profile;
+    const defaultCategory = meta.categories.findIndex(c => c.color === 'grey');
+
+    for (const thread of threads) {
+      const { stackTable, frameTable } = thread;
+      stackTable.category = [];
```

Comment:
same comment as another one earlier: because we know what the final length is, we could as well specify it already.